python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -61,6 +62,47 @@ def _load_genetic_maps_from_feather(map_path: str) -> dict[str, pd.DataFrame]:
         raise e
 
 
+def convert_loci_to_query_format(score_loci: set) -> str:
+     """
+    Convert a set of loci from the format 'chr10:105612479:G:T' to '(chrom:chr10 pos:105612479 inputRef:G alt:T)'
+    and separate them by '||' in order to issue queries with them.
+    """
+    finalized_loci = []
+    for locus in score_loci:
+        chrom, pos, inputRef, alt = locus.split(':')
+        single_query = f"(chrom:{chrom} pos:{pos} inputRef:{inputRef} alt:{alt})"
+        finalized_loci.append(single_query)
+    allele_frequency_filters = [
+        "(gnomad.genomes.AF_afr:>0)",
+        "(gnomad.genomes.AF_amr:>0)",
+        "(gnomad.genomes.AF_eas:>0)",
+        "(gnomad.genomes.AF_nfe:>0)",
+        "(gnomad.genomes.AF_sas:>0)"
+    ]
+    full_query = allele_frequency_filters + finalized_loci
+    return f'"{" || ".join(full_query)}"'
+
+    
+def _extract_af_and_loci_overlap(score_loci: set, target_dataset_index: str, user: str) -> pd.DataFrame:
```

Comment:
user is CachedAuth

Reply:
and then we want to take an optional opensearch_cluster_config, which is a dict

Reply:
and both user and opensearch_cluster_config should be optional; we just need to check that one is not None (and has values)